server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -303,7 +305,8 @@ public GroupShardsIterator<ShardIterator> activePrimaryShardsGrouped(String[] in
                 }
             }
         }
-        return GroupShardsIterator.sortAndCreate(set);
+        CollectionUtil.timSort(set);
```

Comment:
I did check that you have the timSort in all the places where we had it before, but I guess it would be easier to have a single method where we do it? similar to the previous sortAndCreate? I guess you were not sure where to put that? Maybe add it as a static method to ShardIterator ?

Reply:
This was actually on purpose :D I had the same thought, but then I realized something ... this sorting likely isn't necessary in all cases at this point, but also this sorting is one of the reasons for why we cannot iteratively compute target shards because we need to compute everything up-front.
-> I'd leave it explicitly to not hide what's going on here and make for an easier PR that removes some of the sorting down the line?

Reply:
I see, ok.